src/EditorFeatures/CSharp/SplitStringLiteral/SplitStringLiteralCommandHandler.cs

```diff
@@ -56,18 +56,49 @@ public bool ExecuteCommandWorker(ReturnKeyCommandArgs args)
             var spans = textView.Selection.GetSnapshotSpansOnBuffer(subjectBuffer);
 
             // Don't split strings if there is any actual selection.
-            if (spans.Count == 1 && spans[0].IsEmpty)
+            // We must check all spans to account for multi-carets.
+            if (!spans.IsEmpty() && spans.All(s => s.IsEmpty))
             {
                 var caret = textView.GetCaretPoint(subjectBuffer);
                 if (caret != null)
                 {
-                    // Quick check.  If the line doesn't contain a quote in it before the caret,
-                    // then no point in doing any more expensive synchronous work.
-                    var line = subjectBuffer.CurrentSnapshot.GetLineFromPosition(caret.Value);
-                    if (LineContainsQuote(line, caret.Value))
+                    // First, we need to verify that we are only working with string literals.
+                    // Otherwise, let the editor handle all carets.
+                    for (var spanIndex = 0; spanIndex < spans.Count; spanIndex++)
                     {
-                        return SplitString(textView, subjectBuffer, caret.Value);
+                        var spanStart = spans[spanIndex].Start;
+                        var line = subjectBuffer.CurrentSnapshot.GetLineFromPosition(spanStart);
+                        if (!LineContainsQuote(line, spanStart))
+                        {
+                            return false;
+                        }
+                    }
+
+                    // We now go through the verified string literals and split each of them.
```

Comment:
=-o

Out of curiosity, why do this route?  Why not just disable the feature if this is a multi-caret scenario?

Reply:
When a user has multiple carets in various string literals and presses enter, I feel like they would typically expect the same behavior as when splitting a point in a single string literal.

Splitting them as non-strings (the existing functionality) doesn't seem like it would be desired in the majority of scenarios, at least from my perspective. Maybe @sharwell has more insight on this?

Reply:
i'm not cetain what i would expect in this case.  :)   if this works, great!

Reply:
@CyrusNajmabadi Sorry for the very late reply. A very useful scenario is if you have a string with multiple embedded \n. You can then simply select the first one, add carets with Ctrl+Alt+'.', Right, Enter, and voila: properly formatted and indented to more properly reflect how it will be displayed at runtime.